app/modes/library/agent.py

```
@@ -0,0 +1,490 @@
+    if not progress:
+        # Create new progress record
+        progress = BookGenerationProgress(
+            book_id=book_id,
+            user_id=user_id,
+            status='pending',
+            total_topics=total_topics,
+            current_topic_index=0,
+            total_chapters=0,  # Will be calculated during generation
+            completed_chapters=0,
+            current_message='Initializing generation...',
+            started_at=datetime.datetime.utcnow()
```

Ruff (0.15.1)

[warning] `datetime.datetime.utcnow()` used

(DTZ003)